from Common.Errors import InappropriateArgsError
from GUI.solutionView import SolutionView
from unittest import TestCase
import tkinter


class AskIfCorrectViewTester(TestCase):
    def setUp(self):
        self.incorrect_field = [
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
        ]
        self.correct_field = [
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
            [1, 0, 2, 0, 0, 0, 5, 3, 4],
        ]
        self.view = SolutionView(self.correct_field, tkinter.Tk, text="Good job")

    def test_incorrect_init(self):
        self.assertRaises(InappropriateArgsError, SolutionView, self.incorrect_field, tkinter.Tk)
        self.assertRaises(InappropriateArgsError, SolutionView, self.incorrect_field, tkinter.Tk, 1)

    def test_correct_init(self):
        self.assertEqual(self.view.info_label["text"], "Good job")
